spec/requests/admin/supplier_users_spec.rb

```diff
@@ -0,0 +1,194 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe "Admin::SupplierUsers", type: :request do
+  let(:lead_provider) { create(:lead_provider) }
+  let(:delivery_partner) { create(:delivery_partner) }
+  let(:full_name) { Faker::Name.name }
+  let(:email) { Faker::Internet.email }
+
+  before do
+    admin_user = create(:user, :admin)
+    sign_in admin_user
+  end
+
+  describe "GET /admin/suppliers/users" do
+    it "renders the index template" do
+      get "/admin/suppliers/users"
+
+      expect(response).to render_template(:index)
+    end
+  end
+
+  describe "GET /admin/suppliers/users/new" do
+    it "renders the new template" do
+      get "/admin/suppliers/users/new"
+
+      expect(response).to render_template(:new)
+    end
+  end
+
+  describe "POST /admin/suppliers/new" do
+    it "redirects to the supplier type page" do
+      when_i_choose_a_supplier(lead_provider)
+
+      expect(response).to redirect_to("/admin/suppliers/users/new/user-details")
+    end
+
+    it "Sets the correct lead provider" do
+      when_i_choose_a_supplier(lead_provider)
+
+      # Then
+      given_i_have_entered_user_details(full_name, email)
+      given_i_have_confirmed_my_choices
+
+      expect(User.find_by_email(email).lead_provider).to eq lead_provider
+    end
+
+    it "Sets the correct delivery partner" do
+      when_i_choose_a_supplier(delivery_partner)
+
+      # Then
+      given_i_have_entered_user_details(full_name, email)
+      given_i_have_confirmed_my_choices
+
+      expect(User.find_by_email(email).delivery_partner).to eq delivery_partner
+    end
+
+    it "Shows an error if no supplier is chosen" do
+      when_i_choose_a_supplier(nil)
+
+      # Then
+      expect(response).to render_template(:new)
+      expect(response.body).to include("Select one")
+    end
+  end
+
+  describe "GET /admin/suppliers/users/new/user-details" do
+    it "renders the user_details template" do
+      given_i_have_chosen_a_supplier(lead_provider)
+
+      # When
+      get "/admin/suppliers/users/new/user-details"
+
+      # Then
+      expect(response).to render_template(:user_details)
+    end
+  end
+
+  describe "POST /admin/suppliers/users/new/user-details" do
+    before do
+      given_i_have_chosen_a_supplier(lead_provider)
+    end
+
+    it "redirects when user details are entered" do
+      when_i_enter_user_details(full_name, email)
+
+      # Then
+      expect(response).to redirect_to("/admin/suppliers/users/new/review")
+    end
+
+    it "sets the correct user details" do
+      when_i_enter_user_details(full_name, email)
+
+      # Then
```

Comment:
To not repeat yourself too many times - your tests are very clear and easy to read - no need for putting additional comments

Reply:
Boom, PR merged anyway 😂 

I guess Ethan didn't check his email in time...